### DISCUSSION TITLE
can I use a .env var in the form recipient field?

### DISCUSSION THREAD

**Discussion by RyanRoberts** [statamic/cms#10557] [Q&A]
Just a basic Statamic contact form 👉 https://media.yep.digital/kJP4YG0m 

I was wondering if I could change this depending on the environment. I've updated the mail config file so I can access the relevant environment variable, but I'm getting the impression this won't work the way I want.

**Comment by edalzell** [comment]
Don't use the built in email stuff, handle it in your own listener.

However it miiiight work with antlers in there, have you tried? `{{ config:mail:field_that_has_the_email_address }}` ?

**Reply by RyanRoberts** [reply]
Yeah I've tried that, no luck. I'll have a look into your suggestion.

**Comment by martyf** [comment]
Follow Erin's advice here - best to do this in your own listener.

Within the Form and email sending that Statamic does, submitted variables can be used in some config fields, but not things like the config helper in Antlers. Unless I'm missing something too, but we've tried similar things in the past and was just more reliable to have our own listener to handle custom logic.

For example, we've used submitted variables to help tweak the subject for an email - that's able to be done via Antlers, such as `Website Contact from {{ contact_email }}` assuming you have a form field handle with `contact_email`.

Side note: just make sure you're using a registered config variable, not an env variable: when your config is cached, the `env()` helper can't be relied on.
